### ISSUE TITLE
Disribution files for the same version may differ api.pulumi.com

### ISSUE THREAD

**Issue by ejiek** [pulumi/ci-mgmt#874]
### What happened?

I'm sorry for bringing this issue here but it's the best place I've found to report it. It's not a problem with `pulumi-aws` itself but with a way it's distributed.

NixOS fails to build `pulumi-bin` v3.112.0 with pulumi-resource-aws `v6.28.1`

Nix package uses https://api.pulumi.com/releases/plugins/pulumi-resource-aws-v6.28.1-linux-amd64.tar.gz to download the plugin.
We've found that different archive is served for different locations. 2 versions are found so far. Both of them have the same content, but are packaged in a different way.
This leads to different hash sums for the archives which may lead to a nix package build failure due to a checksum mismatch.

Scope might be bigger than nixpkgs since it's not the only place with hash sum validation.

For more details see [nixpkgs issue](https://github.com/NixOS/nixpkgs/issues/300994) linked in Example Section.

### Example

https://github.com/NixOS/nixpkgs/issues/300994

### Output of `pulumi about`

`pulumi about` is not available since pulumi isn't installed yet.

Affected version of pulumi-resource-aws is `v6.28.1` available at https://api.pulumi.com/.

https://api.pulumi.com/releases/plugins/pulumi-resource-aws-v6.28.1-linux-amd64.tar.gz

### Additional context

`6.28.2` seems to be fine.

My goal with this issue is figuring out why this hash sum drift happened and preventing it in the future.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by ejiek** [comment]
@t0yv0 thanks for this clarification! It was quick and very detailed.
I do appreciate it =]

Current situation is more than reasonable. `v6.28.1` hash mismatch looks like an exception. That means that nixpkgs and others are mostly safe and we can close nixpkgs issue.

Nevertheless, it's possible to build a pipeline that does not lead to this situation. Thanks for moving this issue to the ci repo and leaving it open.

Should we leave a broader problem scope well defined for a future evaluation? Reproducible builds, for example.

**Comment by t0yv0** [comment]
That's right. It's an exceptional situation where we should do better, but not an immediate blocker hopefully.
